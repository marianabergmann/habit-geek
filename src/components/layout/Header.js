import React from "react";
import { Link } from "@reach/router";

import "./Header.scss";

function Header(props) {
  const getLinkClass = ({ isCurrent }) => {
    return {
      className: isCurrent ? "nav-item nav-link active" : "nav-item nav-link"
    };
  };
  return (
    <header>
      <nav className="navbar navbar-expand navbar-dark bg-dark header">
        <Link className="navbar-brand" to="/">
          Habit Geek!
        </Link>
        <div className="navbar-nav">
          <Link getProps={getLinkClass} to="/">
            My habits
          </Link>
          <Link getProps={getLinkClass} to="/history">
            History
          </Link>
          <Link getProps={getLinkClass} to="/profile">
            Profile
          </Link>
        </div>
      </nav>
    </header>
  );
}

export default Header;
