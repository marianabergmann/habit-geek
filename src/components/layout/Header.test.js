import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

describe("Header component", () => {
  it("Renders static elements", () => {
    const header = shallow(<Header />);
    expect(header).toMatchSnapshot();
  });
});
