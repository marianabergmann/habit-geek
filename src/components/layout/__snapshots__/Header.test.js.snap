// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders static elements 1`] = `
<header>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark header"
  >
    <ForwardRef
      className="navbar-brand"
      to="/"
    >
      Habit Geek!
    </ForwardRef>
    <div
      className="navbar-nav"
    >
      <ForwardRef
        getProps={[Function]}
        to="/"
      >
        My habits
      </ForwardRef>
      <ForwardRef
        getProps={[Function]}
        to="/history"
      >
        History
      </ForwardRef>
      <ForwardRef
        getProps={[Function]}
        to="/profile"
      >
        Profile
      </ForwardRef>
    </div>
  </nav>
</header>
`;
